test/configuration_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ public:
+  MetadataAgentConfiguration config;
+  static void ParseConfiguration(MetadataAgentConfiguration& config, const std::string configString) {
+    std::stringstream stream(configString);
+    config.ParseConfiguration(stream);
+  }
+};
+
+static void TestDefaultConfig(MetadataAgentConfiguration& config) {
+  EXPECT_EQ("", config.ProjectId());
+  EXPECT_EQ("", config.CredentialsFile());
+  EXPECT_EQ(3, config.MetadataApiNumThreads());
+  EXPECT_EQ(8000, config.MetadataApiPort());
+  EXPECT_EQ(".", config.MetadataApiResourceTypeSeparator());
+  EXPECT_EQ(60, config.MetadataReporterIntervalSeconds());
+  EXPECT_EQ(false, config.MetadataReporterPurgeDeleted());
+  EXPECT_EQ(
+      "https://stackdriver.googleapis.com/"
+      "v1beta2/projects/{{project_id}}/resourceMetadata:batchUpdate",
+      config.MetadataIngestionEndpointFormat());
+  EXPECT_EQ(8*1024*1024, config.MetadataIngestionRequestSizeLimitBytes());
+  EXPECT_EQ("0.1", config.MetadataIngestionRawContentVersion());
+  EXPECT_EQ(60*60, config.InstanceUpdaterIntervalSeconds());
+  EXPECT_EQ("", config.InstanceResourceType());
+  EXPECT_EQ(60, config.DockerUpdaterIntervalSeconds());
+  EXPECT_EQ("unix://%2Fvar%2Frun%2Fdocker.sock/", config.DockerEndpointHost());
+  EXPECT_EQ("1.23", config.DockerApiVersion());
+  EXPECT_EQ("limit=30", config.DockerContainerFilter());
+  EXPECT_EQ(0, config.KubernetesUpdaterIntervalSeconds());
+  EXPECT_EQ("https://kubernetes.default.svc", config.KubernetesEndpointHost());
+  EXPECT_EQ("", config.KubernetesPodLabelSelector());
+  EXPECT_EQ("", config.KubernetesClusterName());
+  EXPECT_EQ("", config.KubernetesClusterLocation());
+  EXPECT_EQ("", config.KubernetesNodeName());
+  EXPECT_EQ(true, config.KubernetesUseWatch());
+  EXPECT_EQ("", config.InstanceId());
+  EXPECT_EQ("", config.InstanceZone());
+}
+
+TEST_F(MetadataAgentConfigurationTest, NoConfigTest) {
+  TestDefaultConfig(config);
+}
+
+TEST_F(MetadataAgentConfigurationTest, EmptyConfigTest) {
+  ParseConfiguration(config, "");
+  TestDefaultConfig(config);
+}
+
+TEST_F(MetadataAgentConfigurationTest, PopulatedConfigTest) {
+  ParseConfiguration(config,
+      "ProjectId: TestProjectId\n"
+      "MetadataApiNumThreads: 13\n"
+      "MetadataReporterPurgeDeleted: true");
+  EXPECT_EQ("TestProjectId", config.ProjectId());
+  EXPECT_EQ(13, config.MetadataApiNumThreads());
+  EXPECT_EQ(true, config.MetadataReporterPurgeDeleted());
+}
+
+TEST_F(MetadataAgentConfigurationTest, CommentSkippedTest) {
+  ParseConfiguration(config,
+      "ProjectId: TestProjectId\n"
+      "#MetadataApiNumThreads: 13\n"
+      "MetadataReporterPurgeDeleted: true");
+  EXPECT_EQ(3, config.MetadataApiNumThreads());
+}
+
+TEST_F(MetadataAgentConfigurationTest, EmptyLineTest) {
+  ParseConfiguration(config,
+      "ProjectId: TestProjectId\n\n\n\n"
```

Comment:
While this is technically correct, it's easier to visualize what's going on if you move each of the `"\n"` to its own line.

Reply:
Agreed, changed.